### PR TITLE
chore: Expand MarkdownLint CI paths

### PR DESCRIPTION
**Pull request by nschonni**

### CODE REVIEW

File: files/en-us/web/javascript/closures/index.md [hunk 1]
```diff
@@ -273,7 +273,7 @@ In the example above, there's a series of nested functions, all of which have ac
 
 ## Creating closures in loops: A common mistake
 
-Prior to the introduction of the [`let` ](/en-US/docs/Web/JavaScript/Reference/Statements/let)keyword in ECMAScript 2015, a common problem with closures occurred when you created them inside a loop. To demonstrate, consider the following example code.
+Prior to the introduction of the [let](/en-US/docs/Web/JavaScript/Reference/Statements/let)keyword in ECMAScript 2015, a common problem with closures occurred when you created them inside a loop. To demonstrate, consider the following example code.
```
Comment (nschonni) [inline files/en-us/web/javascript/closures/index.md:276]:
```suggestion
Prior to the introduction of the [`let`](/en-US/docs/Web/JavaScript/Reference/Statements/let) keyword in ECMAScript 2015, a common problem with closures occurred when you created them inside a loop. To demonstrate, consider the following example code.
```

File: files/en-us/web/javascript/javascript_technologies_overview/index.md [hunk 2]
```diff
@@ -74,6 +74,6 @@ Looking for the [`Document`](/en-US/docs/Web/API/Document "The Document interfac
 - The [`setTimeout`](/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) and [`setInterval`](/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) functions were first specified on the [`Window`](/en-US/docs/Web/API/Window) interface in HTML Standard.
 - [XMLHttpRequest](https://dvcs.w3.org/hg/xhr/raw-file/tip/Overview.html) makes it possible to send asynchronous HTTP requests.
 - The [CSS Object Model](http://dev.w3.org/csswg/cssom/) abstract CSS rules as objects.
-- [WebWorkers ](https://www.whatwg.org/specs/web-workers/current-work/)allows parallel computation.
+- [WebWorkers](https://www.whatwg.org/specs/web-workers/current-work/)allows parallel computation.
```
Comment (nschonni) [inline files/en-us/web/javascript/javascript_technologies_overview/index.md:77]:
```suggestion
- [WebWorkers](https://www.whatwg.org/specs/web-workers/current-work/) allows parallel computation.
```

File: files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ tags:
 ---
 {{jsSidebar("Errors")}}
 
-The JavaScript exception "a declaration in the head of a for-of loop can't have an initializer" occurs when the head of a[ for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression such as |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops.
+The JavaScript exception "a declaration in the head of a for-of loop can't have an initializer" occurs when the head of a[for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression such as |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops.
```
Comment (nschonni) [inline files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md:13]:
```suggestion
The JavaScript exception "a declaration in the head of a for-of loop can't have an initializer" occurs when the head of a [for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression such as |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops.
```

File: files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md [hunk 4]
```diff
@@ -26,7 +26,7 @@ SyntaxError: for-of loop variable declaration may not have an initializer. (Chro
 
 ## What went wrong?
 
-The head of a[ for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression. That is, a variable is declared and assigned a value |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops. You might want a [`for`](/en-US/docs/Web/JavaScript/Reference/Statements/for) loop that does allow an initializer.
+The head of a[for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression. That is, a variable is declared and assigned a value |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops. You might want a [`for`](/en-US/docs/Web/JavaScript/Reference/Statements/for) loop that does allow an initializer.
```
Comment (nschonni) [inline files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md:29]:
```suggestion
The head of a [for...of](/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop contains an initializer expression. That is, a variable is declared and assigned a value |`for (var i = 0 of iterable)`|. This is not allowed in for-of loops. You might want a [`for`](/en-US/docs/Web/JavaScript/Reference/Statements/for) loop that does allow an initializer.
```

File: files/en-us/web/javascript/reference/global_objects/atomics/index.md [hunk 5]
```diff
@@ -128,6 +128,6 @@ Atomics.notify(int32, 0, 1);
 - {{jsxref("ArrayBuffer")}}
 - [JavaScript typed arrays](/en-US/docs/Web/JavaScript/Typed_arrays)
 - [Web Workers](/en-US/docs/Web/API/Web_Workers_API)
-- [parlib-simple ](https://github.com/lars-t-hansen/parlib-simple)– a simple library providing synchronization and work distribution abstractions.
+- [parlib-simple](https://github.com/lars-t-hansen/parlib-simple)– a simple library providing synchronization and work distribution abstractions.
```
Comment (nschonni) [inline files/en-us/web/javascript/reference/global_objects/atomics/index.md:131]:
```suggestion
- [parlib-simple](https://github.com/lars-t-hansen/parlib-simple) – a simple library providing synchronization and work distribution abstractions.
```

File: files/en-us/web/javascript/reference/global_objects/sharedarraybuffer/index.md [hunk 6]
```diff
@@ -134,6 +134,6 @@ gl.bufferData(gl.ARRAY_BUFFER, sab, gl.STATIC_DRAW);
 - {{jsxref("ArrayBuffer")}}
 - [JavaScript typed arrays](/en-US/docs/Web/JavaScript/Typed_arrays)
 - [Web Workers](/en-US/docs/Web/API/Web_Workers_API)
-- [parlib-simple ](https://github.com/lars-t-hansen/parlib-simple)– a simple library providing synchronization and work distribution abstractions.
+- [parlib-simple](https://github.com/lars-t-hansen/parlib-simple)– a simple library providing synchronization and work distribution abstractions.
```
Comment (nschonni) [inline files/en-us/web/javascript/reference/global_objects/sharedarraybuffer/index.md:137]:
```suggestion
- [parlib-simple](https://github.com/lars-t-hansen/parlib-simple) – a simple library providing synchronization and work distribution abstractions.
```

File: files/en-us/web/javascript/reference/statements/empty/index.md [hunk 7]
```diff
@@ -26,7 +26,7 @@ The empty statement is a semicolon (`;`) indicating that no statement will
 be executed, even if JavaScript syntax requires one.
 
 The opposite behavior, where you want multiple statements, but JavaScript only allows a
-single one, is possible using a[ block statement](/en-US/docs/Web/JavaScript/Reference/Statements/block),
+single one, is possible using a[block statement](/en-US/docs/Web/JavaScript/Reference/Statements/block),
```
Comment (nschonni) [inline files/en-us/web/javascript/reference/statements/empty/index.md:29]:
```suggestion
single one, is possible using a [block statement](/en-US/docs/Web/JavaScript/Reference/Statements/block),
```

**Review by nschonni — COMMENTED** [review]
Missed whitespace fixups

**Comment by Rumyra** [comment]
I assume this is linting the new markdown docs - @wbamberg would you be able to take a look, it all seems fine to me but it would be good to get another pair of eyes. This changes `markdown-lint.yml` and I guess that's why tests have failed and I'm probably going to have to force merge this. Thanks!

**Comment by wbamberg** [comment]
Thanks for this PR @nschonni ! Could you please add a description or a comment explaining what it's doing?

**Comment by nschonni** [comment]
Sorry, I guess I missed that when I opened it.
The original Markdownlint job was scoped to just the project admin files.
This expands it out to include all files as they are converted to Markdown. There are some rules that are currently failing and may never be fixed, so they've been disabled in `.markdownlint.json`. Many rules can now by auto-fixed with `markdownlint . --fix` after enabling the rule in the config.

**Comment by nschonni** [comment]
Could add a problem matcher to add inline comments to the PR here, or add it later

**Comment by wbamberg** [comment]
Thanks @nschonni ! I will take a look at this but it might take me a little time :).

**Comment by Ryuno-Ki** [comment]
@nschonni Do you know someone with Admin privileges to take care of the failing check in the meantime?

**Comment by nschonni** [comment]
@Ryuno-Ki think it needs to be someone in https://github.com/mdn/content/blob/07e5b089606fd361001e79acbbd648b80a22826c/.github/CODEOWNERS#L91

**Comment by Ryuno-Ki** [comment]
https://github.com/mdn/content/blob/main/REVIEWING.md#topic-review-owners says

> Note that changes to any content areas not explicitly listed below will be handled by the @core-yari-content team, which currently consists of @Rumyra.

**Comment by Ryuno-Ki** [comment]
So @Rumyra might know, who that would be for dev (instead of content).

**Comment by Rumyra** [comment]
I'll wait for @wbamberg to have a closer look, then check this out 👍

**Comment by nschonni** [comment]
Ping @wbamberg

**Review by wbamberg — COMMENTED** [review]
Thanks for this PR, and sorry to be so slow looking at it. I've been deep in some thickets of ancient HTML.

* as far as I can tell this works just fine to enable markdownlint for mdn/content
* I think enabling markdownlint for mdn/content is a good idea
* the rules seem sensible and compatible with the converted Markdown we have (which is post-Prettier). I couldn't find any enabled rules which I disagreed with
* I had a few questions, which are all about rules I'd love us to be able to enable. But I understand we might not want to fix up the content yet. Maybe we should file some issues like "Enable markdownlint MD001"?

I think this will need some documentation in the project README so contributors don't get frustrated by build failures.

I'd also like to hear wider feedback from people not just about the specific rules to enforce but about our general policy for which CI tools we should be running for mdn/content.

File: .markdownlint.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{
+  "default": true,
+  "MD001": false,
```
[the remark this answers is not in the document]
Comment (nschonni) [reply .markdownlint.json:3]:
```
files/en-us/web/javascript/guide/details_of_the_object_model/index.md:136 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time 
[Expected: h3; Actual: h4]
files/en-us/web/javascript/guide/using_promises/index.md:306 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h4]
files/en-us/web/javascript/memory_management/index.md:192 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h4]
files/en-us/web/javascript/reference/functions/rest_parameters/index.md:52 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h3; Actual: h4]
files/en-us/web/javascript/reference/global_objects/object/create/index.md:159 MD001/heading-increment/header-increment Heading levels should only increment by one level at a 
time [Expected: h3; Actual: h4]
```

File: .markdownlint.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  "MD004": {
+    "style": "dash"
+  },
+  "MD010": false,
+  "MD013": false,
+  "MD024": false,
+  "MD026": false,
```
[the remark this answers is not in the document]
Comment (nschonni) [reply .markdownlint.json:10]:
```
files/en-us/web/javascript/data_structures/index.md:129:43 MD026/no-trailing-punctuation Trailing punctuation in heading [Punctuation: '!']
files/en-us/web/javascript/reference/functions/rest_parameters/index.md:200:65 MD026/no-trailing-punctuation Trailing punctuation in heading [Punctuation: '.']
files/en-us/web/javascript/reference/global_objects/eval/index.md:95:20 MD026/no-trailing-punctuation Trailing punctuation in heading [Punctuation: '!']
```

File: .markdownlint.json [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  "MD028": false,
+  "MD030": false,
+  "MD033": false,
+  "MD036": false,
+  "MD037": false,
+  "MD040": false,
+  "MD042": false,
```
[the remark this answers is not in the document]
Comment (nschonni) [reply .markdownlint.json:17]: Mostly just needs some escaping of the square backets
```

File: .markdownlint.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  "MD045": false,
```
[the remark this answers is not in the document]
Comment (nschonni) [reply .markdownlint.json:18]:
```
files/en-us/web/javascript/guide/details_of_the_object_model/index.md:116 MD045/no-alt-text Images should have alternate text (alt text)
files/en-us/web/javascript/guide/details_of_the_object_model/index.md:244 MD045/no-alt-text Images should have alternate text (alt text)
files/en-us/web/javascript/guide/introduction/index.md:94 MD045/no-alt-text Images should have alternate text (alt text)
files/en-us/web/javascript/reference/global_objects/math/index.md:151 MD045/no-alt-text Images should have alternate text (alt text)
files/en-us/web/javascript/reference/global_objects/promise/index.md:36 MD045/no-alt-text Images should have alternate text (alt text)
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks for adding the lists of files which need updates for us to enable those rules! I will file issues to enable them, but think we should wait until all the content is Markdownified before we try to enable them. Otherwise it's just another barrier to getting into Markdown.

I do think before we can merge this we should have an easy way for people to run the linter locally, and document that in the contribution docs. We do something similar for BCD (https://github.com/mdn/browser-compat-data/blob/main/docs/testing.md#validate-the-data) and interactive examples (https://github.com/mdn/interactive-examples/blob/master/JS-Example-Style-Guide.md#javascript-coding-style). What do you think?

**Comment by nschonni** [comment]
> I will file issues to enable them, but think we should wait until all the content is Markdownified before we try to enable them.

Yes, those are a little more difficult to automate/fix, but I think the other rules are good to have during the convesion, as they can pick up potential rendering issues as the files are converted

**Comment by nschonni** [comment]
> I do think before we can merge this we should have an easy way for people to run the linter locally, and document that in the contribution docs.

I kept this isolated to the CI, but there is no reason it couldn't be added to the actual package.json as an extra target. I'm not sure the best place for the documentation though. It seems like most of it is in the README.md currently

**Comment by Elchi3** [comment]
@ddbeck and I talked about this stalled PR. I opened https://github.com/mdn/content/pull/7146 to discuss next steps and I'm closing here. Thanks for your input!

**Pull request closed without merging**